src/server.rs

```diff
@@ -29,6 +29,38 @@ impl Server {
         })
     }
 
+    /// Creates a server configured to serve files from a given directory on
+    /// a given ip_address and a random port.
+    /// It is attempted to find a free port 16 times before giving up with `AddrInUse`.
+    /// On success the chosen port and the new `Server` instance are returned.
+    pub fn new_with_random_port<A: ToSocketAddrs, P: AsRef<Path>>(
+        ip_addr: &str,
+        serve_from: P,
+    ) -> Result<(u16, Self)> {
```

Comment:
```suggestion
    pub fn random_port<A: AsRef<str>, P: AsRef<Path>>(
        ip_addr: A,
        serve_from: P,
    ) -> Result<Self> {
```

I suggested a change to the return value because I don't think the caller cares _which_ port has been chosen.

Reply:
Wouldn't the caller need to know the port in order to connect to the server?

Reply:
Ah, you're absolutely right, good catch! Let's go with your return type.